src/utilities/convex_hull.zig

```diff
@@ -202,8 +250,52 @@ pub const ConvexHull = struct {
         return hull;
     }
 
-    /// Auxiliary function to add a new `point` to a given `hull` of the convex hull. The function uses
-    /// the given `turn` to correctly add the new point.
+    /// Helper function to get all points sorted as an `ArrayList(DiscretePoint)`. The function
+    /// merges the lower and upper hulls, which are already sorted, in O(N) time. If the
+    /// `ConvexHull` is empty, an empty ArrayList is returned.
+    pub fn getAllPointsSorted(self: *ConvexHull) !ArrayList(DiscretePoint) {
+        // Initialize the result list with enough capacity for both hulls
+        var all_points = try ArrayList(DiscretePoint).initCapacity(
+            self.allocator,
+            self.lower_hull.items.len + self.upper_hull.items.len - 2,
```

Comment:
Maybe add a comment why space for `- 2` `DiscretePoints` is allocated.

Reply:
sure, I had that comment somewhere else so I just repeated there.